### PR TITLE
fix(FEC-8572): change the way the player compute text tracks font size

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

instead of using CSS class to override the vtt style, we will style each caption according to the settings.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1116,9 +1116,16 @@ export default class Player extends FakeEventTarget {
         sheet.insertRule(`video.${ENGINE_CLASS_NAME}::cue { ${style.toCSS()} }`, 0);
       }
       this._textStyle = style;
-      for (let i = 0; i < this._activeTextCues.length; i++) {
-        this._activeTextCues[i].hasBeenReset = true;
+      let videoEl = this.getVideoElement();
+      if (videoEl) {
+        let activeTextTrack = Array.from(videoEl.textTracks).find(track => track && track.mode !== 'disabled');
```
Comment (itaykinnrot) [inline src/player.js]: Need to move to engine level

Comment (odedhutzler) [reply src/player.js]: moved + added in playkit-flash

File: src/player.js [hunk 2]
```diff
@@ -1114,11 +1114,14 @@ export default class Player extends FakeEventTarget {
 
     try {
       if (this._config.playback.useNativeTextTrack) {
-        sheet.insertRule(`#${this._playerId}  video.${ENGINE_CLASS_NAME}::cue { ${style.toCSS()} }`, 0);
-      } else {
-        sheet.insertRule(`#${this._playerId} .${SUBTITLES_CLASS_NAME} > div > div > div { ${style.toCSS()} }`, 0);
+        sheet.insertRule(`#${this._playerId}video.${ENGINE_CLASS_NAME}::cue { ${style.toCSS()} }`, 0);
       }
       this._textStyle = style;
+      if (this._engine) {
```
Comment (OrenMe) [inline src/player.js]: this should be in the else clause - we don't want to run it for self rendered text tracks

File: src/engines/html5/html5.js [hunk 3]
```diff
@@ -907,6 +908,19 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     this.dispatchEvent(new FakeEvent(CustomEventType.TEXT_CUE_CHANGED, {cues: activeCues}));
   }
 
+  /**
+   * set hasBeenReset to true for all the cues. (use case: when cues should be recalculated for display)
+   * @returns {void}
+   */
+  resetAllCues(): void {
+    let activeTextTrack = Array.from(this._el.textTracks).find(track => track && track.mode !== 'disabled');
```
Comment (OrenMe) [inline src/engines/html5/html5.js]: what about the tracks that already have cues but are now in disabled mode?

Comment (odedhutzler) [reply src/engines/html5/html5.js]: they will be reset when the user will select the textTrack which they belong to

**Pull request merged**